src/GraphQL/Types/Scalars/ScalarGraphType.cs

```diff
@@ -116,41 +129,50 @@ public virtual bool IsValidDefault(object value)
-
-            try
-            {
-                var serialized = Serialize(value);
-                return serialized switch
-                {
-                    bool b => new BooleanValue(b),
-                    byte b => new IntValue(b),
-                    sbyte sb => new IntValue(sb),
-                    short s => new IntValue(s),
-                    ushort us => new IntValue(us),
-                    int i => new IntValue(i),
-                    uint ui => new LongValue(ui),
-                    long l => new LongValue(l),
-                    ulong ul => new BigIntValue(ul),
-                    BigInteger bi => new BigIntValue(bi),
-                    decimal d => new DecimalValue(d),
-                    float f => new FloatValue(f),
-                    double d => new FloatValue(d),
-                    string s => new StringValue(s),
-                    null => null,
-                    _ => throw new System.NotImplementedException($"Please override the '{nameof(ToAST)}' method of the '{GetType().Name}' scalar to support this operation.")
-                };
-            }
-            catch
+            var serialized = Serialize(value);
+            return serialized switch
             {
-                return null;
-            }
+                bool b => new BooleanValue(b),
+                byte b => new IntValue(b),
+                sbyte sb => new IntValue(sb),
+                short s => new IntValue(s),
+                ushort us => new IntValue(us),
+                int i => new IntValue(i),
+                uint ui => new LongValue(ui),
+                long l => new LongValue(l),
+                ulong ul => new BigIntValue(ul),
+                BigInteger bi => new BigIntValue(bi),
+                decimal d => new DecimalValue(d),
+                float f => new FloatValue(f),
+                double d => new FloatValue(d),
+                string s => new StringValue(s),
+                null => new NullValue(),
+                _ => throw new System.NotImplementedException($"Please override the '{nameof(ToAST)}' method of the '{GetType().Name}' scalar to support this operation.")
+            };
+        }
+
+        internal object ThrowLiteralConversionError(IValue input)
+        {
+            throw new InvalidOperationException($"Unable to convert '{input}' literal from AST representation to the scalar type '{Name}'");
+        }
+
+        // this is often called for serialization errors, since Serialize calls ParseValue by default
+        // also may be called for serialization errors during ToAST, since ToAST calls Serialize by default, which by default calls ParseValue
+        internal object ThrowValueConversionError(object value)
+        {
+            throw new InvalidOperationException($"Unable to convert '{value}' to the scalar type '{Name}'");
+        }
+
+        // also may be called for serialization errors during ToAST, since ToAST calls Serialize by default
```

Comment:
```suggestion
        // also may be called by ScalarGraphType descendants for serialization errors during ToAST, since ToAST calls Serialize by default
```

Since "base" `Serialize` calls `ParseValue` which should not call `ThrowSerializationError`.